### PR TITLE
[SMALLFIX] Avoid retrying unrecoverable alluxio exceptions

### PR DESCRIPTION
**Pull request by aaudiber**
All RPCs can potentially throw runtime exceptions which get converted to AlluxioTExceptions. We should handle this by propagation instead of retrying.

### CODE REVIEW

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -297,6 +298,8 @@ protected synchronized InetSocketAddress getAddress() {
         return rpc.call();
       } catch (ThriftIOException e) {
         throw new IOException(e);
+      } catch (AlluxioTException e) {
+        throw Throwables.propagate(AlluxioException.from(e));
```
Comment (apc999) [inline core/common/src/main/java/alluxio/AbstractClient.java]: Q: no `AlluxioTException` that is recoverable?

Comment (apc999) [inline core/common/src/main/java/alluxio/AbstractClient.java]: also, please javadoc this (unrecoverability) in the catch block

Comment (aaudiber) [inline core/common/src/main/java/alluxio/AbstractClient.java]: We don't currently have any way to differentiate between recoverable and unrecoverable Alluxio exceptions, but I don't think we currently throw any which make sense to retry.

I've updated the javadoc to make it clear that we only retry for thrift-level exceptions.

**Comment by yupeng9** [comment]
LGTM. Thanks for the improvement.

**Pull request merged**